sled-agent/src/rack_setup/service.rs

```diff
@@ -207,12 +263,35 @@ impl ServiceInner {
             )
             .await?;
         }
+
+        let mut records = HashMap::new();
+        for dataset in datasets {
+            records
+                .entry(dataset.srv())
+                .or_insert_with(Vec::new)
```

Comment:
I was going to suggest `.or_default()` instead of `.or_insert_with(..)`, but that would require putting a type annotation on `records`. I might still do that, but that puts me much more on the fence. Up to you.

Reply:
Done in c53a0bf3c48569611d98839117c62a5f9ed3bf62